src/argilla_llama_index/llama_index_handler.py

```diff
@@ -213,49 +125,121 @@ def __init__(
+                            number_of_retrievals
+                        )
+                        required_context_questions = self._add_context_questions(
+                            number_of_retrievals
+                        )
+                        existing_fields = [
+                            field.to_local() for field in self.dataset.fields
+                        ]
+                        existing_questions = [
+                            question.to_local() for question in self.dataset.questions
+                        ]
+                        # If the required fields and questions do not match with the existing ones, update the dataset and upload it again with "-updated" added to the name
+                        if (
+                            all(
+                                element in existing_fields
+                                for element in required_context_fields
+                            )
+                            == False
+                            or all(
+                                element in existing_questions
+                                for element in required_context_questions
+                            )
+                            == False
+                        ):
+                            local_dataset = self.dataset.pull()
+
+                            fields_to_pop = []
+                            for index, field in enumerate(local_dataset.fields):
+                                if field.name.startswith("retrieved_document_"):
+                                    fields_to_pop.append(index)
+                                    fields_to_pop.sort(reverse=True)
+                            else:
+                                for index in fields_to_pop:
+                                    local_dataset.fields.pop(index)
+
+                            questions_to_pop = []
+                            for index, question in enumerate(local_dataset.questions):
+                                if question.name.startswith(
+                                    "rating_retrieved_document_"
+                                ):
+                                    questions_to_pop.append(index)
+                                    questions_to_pop.sort(reverse=True)
+                            else:
+                                for index in questions_to_pop:
+                                    local_dataset.questions.pop(index)
+
+                            for field in required_context_fields:
+                                local_dataset.fields.append(field)
+                            for question in required_context_questions:
+                                local_dataset.questions.append(question)
+                            self.dataset = local_dataset.push_to_argilla(
+                                self.dataset_name + "-updated"
+                            )
 
         except Exception as e:
             raise FileNotFoundError(
```

Comment:
we can simplifgy this and make the error more specific to the usecase also in other places.

Reply:
I'd actually say that this exception is so general is misleading. If we remove it, we really only have to manually handle the error of not finding any file in the init, the rest is handled through Argilla. Should I cover just this case?